Allow to create/update/delete formula mappings

mestachs opened the pull request:
requires api changes : https://github.com/BLSQ/orbf2/pull/221

![formula-mapping](https://user-images.githubusercontent.com/371692/103529799-9823cf80-4e86-11eb-901f-1acd20b37d86.gif)

mestachs committed:
Allow to create/update/delete formula mappings

pjaspers commented on src/components/Formula/TopicBasedFormulas.js:
```diff
@@ -83,62 +84,181 @@ const Dhis2ElementDetails = ({ dhis2Object }) => {
+    if (cell.value.formulaMapping == undefined) {
+      // CREATE
+      try {
+        const resp = await externalApi()
```
Would move this to a single method `createOrUpdate(cell_value)`, I think only the URL needs to change between the two of them.

pjaspers commented on src/components/Shared/DataElementComboAutocomplete.js:
```diff
@@ -0,0 +1,100 @@
+
+    externalApi()
+      .errorType("json")
+      .url(`/de_cocs`)
```
(we could use ' instead of `)

pjaspers commented on src/actions/api.js:
```diff
@@ -25,6 +26,7 @@ export const externalApi = () => {
     "Accept-Language": i18n.language,
     Accept: "application/vnd.api+json;version=2",
     "X-Token": finalToken,
+    "X-Dhis2UserId": user.id,
```
Nice

pjaspers reviewed (approved):
Haven't tried to run it locally (yet), but looks like a very welcome update.

mestachs committed:
extract components, allow to map formula for non topic related formulas

mestachs committed:
reformat

mestachs commented:
Extracted the component and reused it in the "non topic" based formulas mapping

![formula-mapping-2](https://user-images.githubusercontent.com/371692/103567171-962c3180-4ec3-11eb-9a7b-7c7a92ec0296.gif)

for wretch the method differs (put vs post), I've extracted the payload and page reload (need to find something better, but I'm not a redux guru)

I'll try to check for user role "hesabu admin" to enable or not the edition.

mestachs committed:
Fix test for initial state for api

mestachs committed:
Add hesabu admin role check

mestachs committed:
Remove debugger

mestachs committed:
Inverted translation

mestachs commented:
now the screen will stay in readonly if the user doesn't have userRole "Hesabu Admin".

the role can created via
![image](https://user-images.githubusercontent.com/371692/103632985-64f44580-4f45-11eb-8b53-1cd6c71a3cb3.png)
